pkg/identity/identitysync/okta.go

```diff
@@ -155,7 +162,7 @@ func (o *OktaSync) ListUsers(ctx context.Context) ([]identity.IDPUser, error) {
 
 	// convert all Okta users to internal users
 	for _, u := range users {
-		user, err := o.idpUserFromOktaUser(ctx, u)
+		user, err := o.idpUserFromOktaUser(ctx, u, nil)
```

Comment:
nit, as per my other comment, move the different logic to the caller so `idpUserFromOktaUser` does only one thing and the behaviour does not change based on parameters.
```suggestion
		user, err := o.idpUserFromOktaUser(ctx, u)
		if err != nil {
			return nil, errors.Wrapf(err, "converting okta user %s to internal user", u.Id)
		}
		userGroups, _, err := o.client.User.ListUserGroups(ctx, u.Id)
		if err != nil {
			return u, errors.Wrapf(err, "listing groups for okta user %s", u.Id)
		}
		for _, g := range userGroups {
			user.Groups = append(user.Groups, g.Id)
		}
	}
```